src/docs/development/packages-and-plugins/plugin-api-migration.md

```diff
@@ -169,138 +117,137 @@ but aren't required.
+    }
+    ```
+
+    ```java
+    package io.flutter.plugins.firebase.core;
+
+    import androidx.test.rule.ActivityTestRule;
+    import dev.flutter.plugins.e2e.FlutterRunner;
+    import io.flutter.plugins.firebasecoreexample.EmbeddingV1Activity;
+    import org.junit.Rule;
+    import org.junit.runner.RunWith;
+
+    @RunWith(FlutterRunner.class)
+    public class EmbeddingV1ActivityTest {
+      @Rule
+      public ActivityTestRule<EmbeddingV1Activity> rule =
+          new ActivityTestRule<>(EmbeddingV1Activity.class);
+    }
+    ```
 
-3. Add `e2e` and `flutter_driver` dev_dependencies to
+1. Add `e2e` and `flutter_driver` dev_dependencies to
    `<plugin_name>/pubspec.yaml` and
    `<plugin_name>/example/pubspec.yaml`.
 
-```yaml
-e2e: ^0.2.1
-flutter_driver:
-  sdk: flutter
-```
+    ```yaml
+    e2e: ^0.2.1
+    flutter_driver:
+      sdk: flutter
+    ```
 
-4. Manually register the E2E plugin in `MainActivity.java`
+1. Manually register the E2E plugin in `MainActivity.java`
    alongside any other plugins used by the example app.
 
-```java
-package io.flutter.plugins.packageinfoexample;
-
-import dev.flutter.plugins.e2e.E2EPlugin;
-import io.flutter.embedding.android.FlutterActivity;
-import io.flutter.embedding.engine.FlutterEngine;
-import io.flutter.plugins.packageinfo.PackageInfoPlugin;
-
-public class MainActivity extends FlutterActivity {
-  // TODO(jackson): Remove this once v2 of GeneratedPluginRegistrant
-  // rolls to stable.
-  // https://github.com/flutter/flutter/issues/42694
-  @Override
-  public void configureFlutterEngine(FlutterEngine flutterEngine) {
-    flutterEngine.getPlugins().add(new PackageInfoPlugin());
-    flutterEngine.getPlugins().add(new E2EPlugin());
-  }
-}
-```
+    ```java
+    package io.flutter.plugins.packageinfoexample;
+
+    import dev.flutter.plugins.e2e.E2EPlugin;
+    import io.flutter.embedding.android.FlutterActivity;
+    import io.flutter.embedding.engine.FlutterEngine;
+    import io.flutter.plugins.packageinfo.PackageInfoPlugin;
+
+    public class MainActivity extends FlutterActivity {
+      // TODO(jackson): Remove this once v2 of GeneratedPluginRegistrant
+      // rolls to stable.
+      // https://github.com/flutter/flutter/issues/42694
+      @Override
+      public void configureFlutterEngine(FlutterEngine flutterEngine) {
+        flutterEngine.getPlugins().add(new PackageInfoPlugin());
+        flutterEngine.getPlugins().add(new E2EPlugin());
+      }
+    }
+    ```
 
-4. Update minimum Flutter version of environment in
+1. Update minimum Flutter version of environment in
    `<plugin_name>/pubspec.yaml`. All plugins moving
    forward will set the minimum version to 1.9.1+hotfix.4
    which is the minimum version for which we can guarantee support.
    For example:
 
-```yaml
-environment:
-  sdk: ">=2.0.0-dev.28.0 <3.0.0"
-  flutter: ">=1.9.1+hotfix.4 <2.0.0"
+    ```yaml
+    environment:
+      sdk: ">=2.0.0-dev.28.0 <3.0.0"
+      flutter: ">=1.9.1+hotfix.4 <2.0.0"
```

Comment:
I think this should be the new stable now, 1.12. 1.9 is missing some fixes that will matter in a few edge cases so I think it's better to just recommend stable here.